# Arithmetic and Variables

## Printing

In [1]:
print("Hello, world!")

Hello, world!


Kode ini adalah contoh sederhana dari bahasa pemrograman Python. Perintah print() digunakan untuk menampilkan pesan ke layar. Dalam kasus ini, pesan yang ditampilkan adalah "Hello, world!". Saat Anda menjalankan kode ini, Anda akan melihat output "Hello, world!" di layar.

## Arithmetic

In [2]:
print(1 + 2)

3


Baris kode ini menggunakan operator penjumlahan dalam bahasa pemrograman Python. Operator + digunakan untuk menambahkan dua nilai bersama-sama. Dalam kasus ini, nilai yang ditambahkan adalah 1 dan 2. Hasil dari operasi tersebut adalah 3.

Jadi, saat kode ini dijalankan, hasilnya akan dicetak ke layar, yang merupakan angka 3.

In [3]:
print(9 - 5)

4


Baris kode ini menggunakan operator pengurangan dalam bahasa pemrograman Python. Operator - digunakan untuk mengurangkan satu nilai dari nilai lainnya. Dalam kasus ini, kita mengurangkan 5 dari 9. Hasil dari operasi tersebut adalah 4.

Jadi, saat kode ini dijalankan, hasilnya, yaitu angka 4, akan dicetak ke layar.

In [4]:
print(((1 + 3) * (9 - 2) / 2) ** 2)

196.0


Baris kode ini melakukan serangkaian operasi matematika menggunakan ekspresi yang rumit. Mari kita pecah ekspresi tersebut:

(1 + 3): Pertama, dilakukan penjumlahan antara 1 dan 3, hasilnya adalah 4.

(9 - 2): Kemudian, dilakukan pengurangan antara 9 dan 2, hasilnya adalah 7.

((1 + 3) * (9 - 2) / 2): Setelah itu, hasil dari penjumlahan dan pengurangan sebelumnya (yaitu 4 dan 7) dikalikan dan dibagi dengan 2. Hasilnya adalah 14.

((1 + 3) * (9 - 2) / 2) ** 2: Setelahnya, hasil dari operasi sebelumnya dipangkatkan dengan pangkat 2. Artinya, hasil sebelumnya yaitu 14 dipangkatkan dengan 2, sehingga hasilnya adalah 196.

Jadi, ekspresi ((1 + 3) * (9 - 2) / 2) ** 2 akhirnya akan menghasilkan nilai 196.

## Comments

Komentar adalah bagian dari kode yang kita tulis untuk menjelaskan apa yang dilakukan oleh kode tersebut. Mereka membantu kita dan orang lain memahami tujuan dan logika di balik potongan kode tertentu. Misalnya, jika Anda membuat fungsi atau bagian dari kode yang kompleks, Anda dapat menambahkan komentar untuk menjelaskan bagaimana kode itu bekerja atau mengapa Anda menggunakan pendekatan tertentu.

In [5]:
# Multiply 3 by 2
print(3 * 2)

6


Multiply 3 by 2 adalah komentar yang dimaksudkan untuk memberikan penjelasan tentang apa yang dilakukan oleh kode berikutnya. Komentar dimulai dengan tanda pagar (#) dalam Python. Komentar ini menjelaskan bahwa kita akan mengalikan angka 3 dengan 2.

jika kita menghapus tanda pagar dari sebuah komentar, Python akan menimbulkan kesalahan karena teks di dalam komentar tersebut bukan kode Python yang valid. Python mencoba untuk menafsirkan setiap bagian dari kode sebagai instruksi yang dapat dijalankan, tetapi teks dalam komentar tidak dimaksudkan untuk dieksekusi sebagai kode Python. Oleh karena itu, Python tidak dapat memahami atau menafsirkan teks tersebut jika tanda pagar dihapus.

In [6]:
Multiply 3 by 2

SyntaxError: invalid syntax (3750420471.py, line 1)

## Variables

### Creating Variables

In [7]:
# Create a variable called test_var and give it a value of 4+5
test_var = 4 + 5

# Print the value of test_var
print(test_var)

9


Create a variable called test_var and give it a value of 4+5: Komentar ini menjelaskan bahwa kita akan membuat sebuah variabel yang disebut test_var dan memberikannya nilai hasil penjumlahan dari 4+5.

test_var = 4 + 5: Baris kode ini secara aktual membuat variabel bernama test_var dan memberikannya nilai dari hasil penjumlahan 4 + 5, yang akan menjadi 9. Ini adalah contoh penugasan variabel di Python.

Print the value of test_var: Komentar ini menjelaskan bahwa kita akan mencetak nilai dari variabel test_var.

print(test_var): Baris kode ini adalah perintah print yang mencetak nilai dari variabel test_var. Hasilnya adalah 9, karena itulah nilai yang telah kita berikan ke variabel test_var.

Pemilihan Nama Variabel: Nama variabel harus singkat dan deskriptif agar mudah dipahami.

Persyaratan Nama Variabel:

Tidak boleh ada spasi dalam nama variabel.
Hanya boleh menggunakan huruf, angka, dan garis bawah.
Dimulai dengan huruf atau garis bawah, tidak boleh dengan angka.
Membuat Variabel: Gunakan tanda sama dengan (=) untuk menugaskan nilai ke variabel.

Melihat Nilai Variabel: Gunakan fungsi print() untuk mencetak nilai variabel.

Pemilihan Nama Variabel yang Baik: Praktik konsisten dan pengalaman akan membantu memilih nama variabel yang baik.

### Manipulating Variables

In [8]:
# Set the value of a new variable to 3
my_var = 3

# Print the value assigned to my_var
print(my_var)

# Change the value of the variable to 100
my_var = 100

# Print the new value assigned to my_var
print(my_var)

3
100


Secara umum, ketika Anda mendefinisikan sebuah variabel dalam sebuah sel kode, semua sel kode yang mengikuti juga memiliki akses ke variabel tersebut. Misalnya, disini amenggunakan sel kode berikutnya untuk mengakses nilai dari my_var (dari sel kode di atas) dan test_var (dari sebelumnya dalam tutorial ini).

In [9]:
print(my_var)
print(test_var)

100
9


Sel selanjutnya memberi tahu Python untuk meningkatkan nilai saat ini dari my_var sebesar 3.

Untuk melakukannya, kita masih perlu menggunakan my_var = seperti sebelumnya. Dan juga seperti sebelumnya, nilai baru yang ingin kita berikan ke variabel berada di sebelah kanan tanda =.

In [10]:
# Increase the value by 3
my_var = my_var + 3

# Print the value assigned to my_var
print(my_var)

103


### Using Multiple Variables

In [11]:
# Create variables
num_years = 4
days_per_year = 365 
hours_per_day = 24
mins_per_hour = 60
secs_per_min = 60

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126144000


Seperti yang dihitung di atas, ada 126.144.000 detik dalam empat tahun.

Perlu diperhatikan bahwa memang mungkin untuk melakukan perhitungan ini tanpa menggunakan variabel hanya dengan menggunakan perhitungan langsung seperti 60 * 60 * 24 * 365 * 4, tetapi itu lebih sulit untuk memeriksa apakah perhitungan tanpa variabel memiliki kesalahan, karena kurang mudah dibaca. Ketika kita menggunakan variabel (seperti num_years, days_per_year, dll.), kita dapat lebih baik melacak setiap bagian dari perhitungan dan lebih mudah memeriksa dan memperbaiki kesalahan.

Perlu diperhatikan bahwa sangat berguna untuk menggunakan variabel ketika nilai-nilai dari input dapat berubah. Misalnya, katakanlah kita ingin sedikit meningkatkan perkiraan kami dengan memperbarui nilai jumlah hari dalam setahun dari 365 menjadi 365.25, untuk memperhitungkan tahun kabisat. Kemudian kita dapat mengubah nilai yang ditugaskan ke days_per_year tanpa mengubah variabel lainnya dan mengulang perhitungannya.

In [12]:
# Update to include leap years
days_per_year = 365.25

# Calculate number of seconds in four years
total_secs = secs_per_min * mins_per_hour * hours_per_day * days_per_year * num_years
print(total_secs)

126230400.0


## Debugging

In [13]:
print(hours_per_dy)

NameError: name 'hours_per_dy' is not defined

Ketika Anda melihat pesan kesalahan "NameError" seperti ini, itu berarti Python tidak dapat menemukan variabel yang Anda coba gunakan. Biasanya, hal ini terjadi karena kesalahan dalam pengejaan nama variabel tersebut. Jadi, untuk memperbaiki kesalahan ini, Anda perlu memeriksa bagaimana Anda mengeja nama variabelnya dan pastikan ejaannya benar.

In [14]:
print(hours_per_day)

24


Ini akan mencetak jumlah jam dalam sehari ke layar. Jika variabel hours_per_day telah didefinisikan sebelumnya, maka nilai yang dihasilkan akan ditampilkan.